dist/clouduploader.rb

```diff
@@ -31,57 +97,83 @@ def self.credentials(data)
+      image_name = job_data['image_name'].to_s
+      image_name = calculate_image_name(image_filename) if image_name.empty?
+      uncompressed_file = uncompress(backend_image_file)
+      remote_file_name = File.basename(uncompressed_file)
 
-      credentials = credentials(data)
+      run_command("az login",
+                  "--service-principal -u #{job_data['application_id']} -p #{job_data['application_key']} " \
+                  "--tenant #{job_data['subscription']} --debug",
+                  "Logging in as OBS app")
 
-      Open3.popen2e(
-        'ec2uploadimg',
-        "--description='obs uploader'",
-        '--machine=x86_64',
-        "--name=#{data['ami_name']}",
-        "--region=#{data['region']}",
-        "--secret-key=#{credentials.secret_access_key}",
-        "--access-id=#{credentials.access_key_id}",
-        "--session-token=#{credentials.session_token}",
-        '--verbose',
-        image
-      ) do |_stdin, stdout_stderr, _wait_thr|
-        while line = stdout_stderr.gets
-          STDOUT.write(line)
-        end
-        status = wait_thr.value
-        abort unless status.success?
-      end
+      run_command("az storage container create",
+                  "-n #{job_data['container']} --account-name #{job_data['storage_account']} --debug",
+                  "Creating container at '#{job_data['storage_account']}/#{job_data['container']}'")
+
+      run_command("az storage blob upload",
+                  "--container-name #{job_data['container']} --account-name #{job_data['storage_account']} -f #{uncompressed_file} " \
+                  "-n #{remote_file_name} --debug",
+                  "Uploading image file '#{uncompressed_file}' to a blob")
+
+      source_url = uploaded_image_url(job_data['storage_account'], job_data['container'], remote_file_name)
+      result = run_command("az image create",
+                           "--resource-group #{job_data['resource_group']} --name #{image_name} " \
+                           "--source #{source_url} --os-type #{job_data['os_type']} --debug",
+                           "Creating image '#{image_name}' out of the blob '#{remote_file_name}'")
+      STDOUT.write("#{JSON.parse(result).inspect}\n")
+
+      run_command("az storage blob delete",
+                  "--container-name #{job_data['container']} --account-name #{job_data['storage_account']} -n #{remote_file_name} --debug",
+                  "Deleting")
+
+      run_command("az logout", "", "Logging out")
     end
-  end
 
-  def self.upload(platform, image_path, job_data_file, image_filename)
-    STDOUT.sync = true
-    start = Time.now
+    def self.uncompress(backend_image_file)
+      compressed_file = "#{File.join(Dir.pwd, backend_image_file)}.xz"
+      uncompressed_file = "#{File.join(Dir.pwd, backend_image_file)}.vhd"
+      run_command("cp", "#{backend_image_file} #{compressed_file}", "Copying the file '#{backend_image_file}' to working directory and rename to ensure uncompression")
```

Comment:
Why do you copy it first?

Reply:
IIRC xz doesn't work with symbolic links...

Reply:
Unless you write to stdout...

Reply:
Done, using a symbolic link now instead of copying